scripts/sct_label_utils.py

```diff
@@ -21,711 +21,81 @@
-        """
-        Create an image with labels defined along the spinal cord segmentation (or centerline).
-        Input image does **not** need to be RPI (re-orientation is done within this function).
-        Example:
-        object_define=ProcessLabels(fname_segmentation, coordinates=[coord_1, coord_2, coord_i]), where coord_i='z,value'. If z=-1, then use z=nz/2 (i.e. center of FOV in superior-inferior direction)
-        Returns
-        """
-        # reorient input image to RPI
-        im_rpi = self.image_input.copy().change_orientation('RPI')
-        im_output_rpi = zeros_like(im_rpi)
-        # loop across labels
-        for ilabel, coord in enumerate(self.coordinates):
-            # split coord string
-            list_coord = coord.split(',')
-            # convert to int() and assign to variable
-            z, value = [int(i) for i in list_coord]
-            # update z based on native image orientation (z should represent superior-inferior axis)
-            coord = Coordinate([z, z, z])  # since we don't know which dimension corresponds to the superior-inferior
-            # axis, we put z in all dimensions (we don't care about x and y here)
-            _, _, z_rpi = coord.permute(self.image_input, 'RPI')
-            # if z=-1, replace with nz/2
-            if z == -1:
-                z_rpi = int(np.round(im_output_rpi.dim[2] / 2.0))
-            # get center of mass of segmentation at given z
-            x, y = ndimage.measurements.center_of_mass(np.array(im_rpi.data[:, :, z_rpi]))
-            # round values to make indices
-            x, y = int(np.round(x)), int(np.round(y))
-            # display info
-            sct.printv('Label #' + str(ilabel) + ': ' + str(x) + ',' + str(y) + ',' + str(z_rpi) + ' --> ' + str(value), 1)
-            if len(im_output_rpi.data.shape) == 3:
-                im_output_rpi.data[x, y, z_rpi] = value
-            elif len(im_output_rpi.data.shape) == 2:
-                assert str(z) == '0', "ERROR: 2D coordinates should have a Z value of 0. Z coordinate is :" + str(z)
-                im_output_rpi.data[x, y] = value
-        # change orientation back to native
-        return im_output_rpi.change_orientation(self.image_input.orientation)
-
-    def plan(self, width, offset=0, gap=1):
-        """
-        Create a plane of thickness="width" and changes its value with an offset and a gap between labels.
-        """
-        image_output = msct_image.zeros_like(self.image_input)
-
-        coordinates_input = self.image_input.getNonZeroCoordinates()
-
-        # for all points with non-zeros neighbors, force the neighbors to 0
-        for coord in coordinates_input:
-            image_output.data[:, :, int(coord.z) - width:int(coord.z) + width] = offset + gap * coord.value
-
-        return image_output
-
-    def plan_ref(self):
-        """
-        Generate a plane in the reference space for each label present in the input image
-        """
```

Comment:
I can't find this in the new `labels.py` api file. Was this not used anywhere?

Reply:
Not used anywhere

Reply:
@Drulex Instead of removing them, is possible we still want access to the `diff`, `plan`, and `plan_ref` functions, e.g. for future development? Might be worth checking in with someone who might have used them in the past, e.g. @jcohenadad.

Reply:
A wise man once said:
> if it's not usable nor unit-tested it's just a liability...

credit @zougloub

Reply:
Fair point! :)